### PR TITLE
Improve test coverage

### LINKED ISSUE #4: Extend test coverage
Tests are in place only for the news article content type and a news article page.

Coverage needs to be extended to cover all blocks and the newsroom sub-module.

### PR DESCRIPTION
**Pull request by danchamp**
Adds tests for:

- Creating news article
- Creating newsroom
- News listing
- Featured news
- Creation and use of default news media
- Function of 'Show more' button
- Basic news search

Closes #4

### CODE REVIEW

File: tests/src/Functional/NewsPageTest.php [hunk 1]
```diff
@@ -22,7 +24,12 @@ class NewsPageTest extends BrowserTestBase {
    *
    * @var string
    */
-  protected $profile = 'standard';
+  protected $profile = 'localgov';
```
Comment (andybroomfield) [inline tests/src/Functional/NewsPageTest.php:27]: Is there a particular reason to test using the profile.
I'm not sure its aggreed yet (one for technical group) if we should rely on the profile being installed, That would mean manually setting blocks etc... So its up to you if you want to do it, or just add the profile as a dev dependency.

**Review by andybroomfield — APPROVED** [review]
Overall looks good and tests are passing so would be good to get this added.

File: tests/src/Functional/NewsPageTest.php [hunk 2]
```diff
@@ -77,24 +84,95 @@ public function testNewsFields() {
...
+    $this->drupalGet('news');
     $this->assertText('News article 1');
 
-    $trail = ['' => 'Home'];
-    $trail += ['news' => 'News'];
-    $this->assertBreadcrumb(NULL, $trail);
+    // Add News article 1 to the featured news block.
+    $newsroom->set('localgov_newsroom_featured', $newsArticle->id());
+    $newsroom->save();
+    drupal_flush_all_caches();
```
Comment (andybroomfield) [inline tests/src/Functional/NewsPageTest.php:117]: Is there a reason for the flush all caches? Be good to have a look at the cache tags on that block and remove this. Feels like updating a news featured block shouldn't require a cahce clear, and if removing this causes the test to fail, that indicated something that needs fixing.

File: tests/src/Functional/NewsPageTest.php [hunk 3]
```diff
@@ -77,24 +84,95 @@ public function testNewsFields() {
...
+    // Create default news image entity using example file.
+    $newsMedia = Media::create([
+      'name' => 'Example',
+      'bundle' => 'localgov_news_default_image',
+      'field_media_image' => [
+        'target_id' => $imageFile->id(),
+        'alt' => 'Alternative text',
+      ],
+    ]);
+    $newsMedia->setPublished()->save();
```
Comment (andybroomfield) [inline tests/src/Functional/NewsPageTest.php:147]: Could we also test the select from a list functionalty (if thats a new issue I'll raise it).

File: tests/src/FunctionalJavascript/NewsShowMoreTest.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\Tests\localgov_alert_banner\FunctionalJavascript;
```
Comment (andybroomfield) [inline tests/src/FunctionalJavascript/NewsShowMoreTest.php:3]: Wrong namespace! :-)

File: tests/src/Functional/NewsPageTest.php (hunk 2 above)
Comment (danchamp) [reply tests/src/Functional/NewsPageTest.php:117]: It fails without clearing caches. I've no experience with cache tags but will take a look, good opportunity to learn.

File: tests/src/FunctionalJavascript/NewsShowMoreTest.php (hunk 4 above)
Comment (danchamp) [reply tests/src/FunctionalJavascript/NewsShowMoreTest.php:3]: Bad copy pasta! Will fix.

File: tests/src/Functional/NewsPageTest.php (hunk 3 above)
Comment (danchamp) [reply tests/src/Functional/NewsPageTest.php:147]: That creates the media type item, it's selected from the list a bit further on:

https://github.com/localgovdrupal/localgov_news/blob/a4d6f93cb7ffe46fbb6d14ca26a29ac7414f8fc6/tests/src/Functional/NewsPageTest.php#L155-L160

Or am I misunderstanding?

File: tests/src/Functional/NewsPageTest.php (hunk 2 above)
Comment (andybroomfield) [reply tests/src/Functional/NewsPageTest.php:117]: Ok, we can merge for now, however I'd look at the localgov services for how to do cache tags and build the test without fluch caches. https://github.com/localgovdrupal/localgov_services/blob/7ce43ae55eadfb286405ef144b50ecddaf888d72/src/Plugin/Block/ServicesBlockBase.php#L92

File: tests/src/Functional/NewsPageTest.php (hunk 3 above)
Comment (andybroomfield) [reply tests/src/Functional/NewsPageTest.php:147]: Sorry misunderstanding, can resolve this now.

File: tests/src/Functional/NewsPageTest.php (hunk 2 above)
Comment (danchamp) [reply tests/src/Functional/NewsPageTest.php:117]: Thanks @andybroomfield, I'll add an issue and have a look at that.

File: tests/src/Functional/NewsPageTest.php (hunk 1 above)
Comment (danchamp) [reply tests/src/Functional/NewsPageTest.php:27]: The news media test is the only one that fails when using the standard install profile.

I couldn't get to the bottom of why, and saw that the localgov profile is used in subsites testing so went with that. I'd prefer to understand why that test fails with the standard profile.

**Pull request merged**